app/services/hyrax/embargo_manager.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Manages an embargo for a `Hyrax::Resource`
```

Comment:
Can you say more in this comment about when/how this class is useful? I think I figured out what you're intending but it wasn't immediately obvious to me.

Reply:
could we push this off to a follow up? this class is developing in conjunction with #3870. i suspect it will be easier to document with some usage examples in place.

i've mainly submitted this as-is because of the changes to `ModelTransformer`.